src/index.ts

```diff
@@ -23,27 +23,33 @@ export interface Event {
 
 export type Callback = (error: any, response: Twilio.Response) => void;
 export type HandlerFn = (context: Context, event: Event, callback: Callback) => void;
-
+export type AuthToken = string;
+export type APIKey = {secret: string, key: string};
 /**
  * Validates that the Token is valid
  *
  * @param token        the token to validate
  * @param accountSid   the accountSid
- * @param authToken    the authToken
+ * @param credential    the authToken
  */
-export const validator = async (token: string, accountSid: string, authToken: string): Promise<object> => {
+export const validator = async (token: string, accountSid: string, credential: APIKey | AuthToken): Promise<object> => {
   return new Promise((resolve, reject) => {
     if (!token) {
       reject('Unauthorized: Token was not provided');
       return;
     }
 
-    if (!accountSid || !authToken) {
+    if (!accountSid || !credential) {
       reject('Unauthorized: AccountSid or AuthToken was not provided');
```

Comment:
@ktalebian 
I was going to reword the reject message, how does the following look?
`'Unauthorized: AccountSid or a Credential (AuthToken or APIKey Object) was not provided'`

Reply:
Also I would update the @params on 33 to:
`* @param credential   the authToken or apiKey Object`

if thats looks ok, maybe this is better since it's type is defined above:
`* @param credential   the AuthToken or APIiKey `

it doesn't look like it would be much value to be more JSDoc savvy, but for completeness-sake 
`* @param {(AuthToken|APIKey)}`

Reply:
Do `* @param credential the {@link AuthToken} or {@link APIKey}`